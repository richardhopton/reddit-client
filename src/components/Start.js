import React, { Component } from "react";
import styles from "./styles.css";

export default class Start extends Component {
  constructor() {
    super();
    this.state = {
      subreddit: "oddlysatisfying"
    };
  }

  onHandleChange(event) {
    this.setState({ subreddit: event.target.value });
  }

  handleClick(e) {
    e.preventDefault();
    this.props.history.push(this.state.subreddit);
  }

  render() {
    return (
      <div>
        <div className={styles.header}>
          <div className={styles.headerName}>
            Enter the subreddit and click Go
          </div>
        </div>
        <div className={styles.body}>
          <div className={styles.picker}>
            /r/
            <input
              type="text"
              value={this.state.subreddit}
              onChange={this.onHandleChange.bind(this)}
            />
          </div>
          <button onClick={this.handleClick.bind(this)}>Go</button>
        </div>
      </div>
    );
  }
}
