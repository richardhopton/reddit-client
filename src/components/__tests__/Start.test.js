import React from "react";
import Start from "../Start";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

// Check the component hasn't been accidentally adjusted by using a jest snapshot
test("Start renders expected snapshot", () => {
  const tree = renderer.create(<Start />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Check interactions
const setup = () => {
  const push = jest.fn();
  const history = { push };
  const start = shallow(<Start history={history} />);

  return { push, start };
};

test("Start pushes history with default url on button click", () => {
  const { push, start } = setup();

  start.find("button").simulate("click", { preventDefault: () => {} });

  expect(push).toHaveBeenCalledWith("oddlysatisfying");
});

test("Start pushes history with updated url on button click", () => {
  const { push, start } = setup();

  start.find("input").simulate("change", { target: { value: "news" } });
  start.find("button").simulate("click", { preventDefault: () => {} });

  expect(push).toHaveBeenCalledWith("news");
});
