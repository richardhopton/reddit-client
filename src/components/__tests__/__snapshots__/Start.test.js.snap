// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start renders expected snapshot 1`] = `
<div>
  <div
    className="header"
  >
    <div
      className="headerName"
    >
      Enter the subreddit and click Go
    </div>
  </div>
  <div
    className="body"
  >
    <div
      className="picker"
    >
      /r/
      <input
        onChange={[Function]}
        type="text"
        value="oddlysatisfying"
      />
    </div>
    <button
      onClick={[Function]}
    >
      Go
    </button>
  </div>
</div>
`;
